src/libraries/System.IO.FileSystem/tests/Base/InfoGetSetTimes.cs

```diff
@@ -65,9 +65,11 @@ public void TimesStillSetAfterDelete()
 
             // Deleting shouldn't change any info state
             item.Delete();
+            
             Assert.All(times, time =>
             {
-                Assert.Equal(time.Value, time.Key.Getter(item));
+                // We check for the time update from refresh
+                Assert.NotEqual(time.Key.Getter.Invoke(item), time.Value);
```

Comment:
Hey @stephentoub  I'm actually worried about this one, the time function stores a delegate to the underlying getter, but I don't think this test ever worked anyway as the value being stored is this the datetime from the first function, not a getter method to the underlying datetime.

[Here](https://github.com/dotnet/runtime/blob/master/src/libraries/System.IO.FileSystem/tests/File/GetSetTimes.cs#L68) we're creating the time function, but then just storing the same value type.

So either I'm going crazy, not understanding this or it didn't work and I'd like to look around the code base for more of these situations

Reply:
I'm not understanding the concern.  The original test runs the getter on the item to get the original time and stores it:
```C#
times.Add(timeFunction, timeFunction.Getter(item));
```
It then deletes the file:
```C#
// Deleting shouldn't change any info state
item.Delete();
```
And then it runs the getter again and validates that it hasn't changed from what was stored:
```C#
Assert.Equal(time.Value, time.Key.Getter(item));
```
validating the behavior prior to your PR that Delete doesn't invalidate the instance.  Can you elaborate on what's wrong?

Reply:
@stephentoub I thought that as well, but we're not actually invoking the delegate.

![image](https://user-images.githubusercontent.com/25160653/77866091-86c6dc00-71ff-11ea-9259-296f04954c5d.png)

So it never was refreshing the underlying time. So the original test checked if all values were before utcnow, which they always would have been

Reply:
I'm still not understanding... it's being invoked here:
https://github.com/dotnet/runtime/pull/34229/files/03c68b79de6dda91c9692f3365ab3f57f4fdcb65#diff-de74d261db69a93767732397a82a4f54L70
`Getter` is the delegate.  What am I missing?

Reply:
You're correct, I'm just making a fool of myself on github. I guess this is what happens when I don't go outside for 2 weeks. I apologize for wasting your time

Reply:
Ok :-)